presto-main/src/main/java/com/facebook/presto/catalogserver/RemoteMetadataManager.java

```diff
@@ -0,0 +1,780 @@
+import com.facebook.presto.metadata.Metadata;
+import com.facebook.presto.metadata.MetadataManager;
+import com.facebook.presto.metadata.MetadataUpdates;
+import com.facebook.presto.metadata.NewTableLayout;
+import com.facebook.presto.metadata.OutputTableHandle;
+import com.facebook.presto.metadata.PartitioningMetadata;
+import com.facebook.presto.metadata.ProcedureRegistry;
+import com.facebook.presto.metadata.QualifiedTablePrefix;
+import com.facebook.presto.metadata.ResolvedIndex;
+import com.facebook.presto.metadata.SchemaPropertyManager;
+import com.facebook.presto.metadata.SessionPropertyManager;
+import com.facebook.presto.metadata.TableLayout;
+import com.facebook.presto.metadata.TableLayoutResult;
+import com.facebook.presto.metadata.TableMetadata;
+import com.facebook.presto.metadata.TablePropertyManager;
+import com.facebook.presto.metadata.ViewDefinition;
+import com.facebook.presto.spi.ColumnHandle;
+import com.facebook.presto.spi.ColumnMetadata;
+import com.facebook.presto.spi.ConnectorId;
+import com.facebook.presto.spi.ConnectorMaterializedViewDefinition;
+import com.facebook.presto.spi.ConnectorTableMetadata;
+import com.facebook.presto.spi.Constraint;
+import com.facebook.presto.spi.MaterializedViewStatus;
+import com.facebook.presto.spi.QueryId;
+import com.facebook.presto.spi.SystemTable;
+import com.facebook.presto.spi.TableHandle;
+import com.facebook.presto.spi.TableLayoutFilterCoverage;
+import com.facebook.presto.spi.connector.ConnectorCapabilities;
+import com.facebook.presto.spi.connector.ConnectorOutputMetadata;
+import com.facebook.presto.spi.connector.ConnectorTransactionHandle;
+import com.facebook.presto.spi.function.SqlFunction;
+import com.facebook.presto.spi.security.GrantInfo;
+import com.facebook.presto.spi.security.PrestoPrincipal;
+import com.facebook.presto.spi.security.Privilege;
+import com.facebook.presto.spi.security.RoleGrant;
+import com.facebook.presto.spi.statistics.ComputedStatistics;
+import com.facebook.presto.spi.statistics.TableStatistics;
+import com.facebook.presto.spi.statistics.TableStatisticsMetadata;
+import com.facebook.presto.sql.planner.PartitioningHandle;
+import com.facebook.presto.transaction.TransactionManager;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.util.concurrent.ListenableFuture;
+import io.airlift.slice.Slice;
+
+import javax.inject.Inject;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalLong;
+import java.util.Set;
+
+import static java.util.Objects.requireNonNull;
+
+// TODO : Use thrift to serialize metadata objects instead of json serde on catalog server in the future
+public class RemoteMetadataManager
```

Comment:
I am not sure where a test for this is? May be I am missing something.

Reply:
Yes, so this ties back to my previous PR #17855. Here is the same question and reply. Same applies to the RemoteMetadataManager as the CatalogServer class

This only tests the client. Is it possible to add an e2e test which tests the client and the CatalogServer implementation?

Me : From my discussions about the testing approach I don't think its possible to use the CatalogServer implementation directly as it would need many dependencies as well as the query runner running to be able to use and call. What this approach at least does is has the testing client return a result that the CatalogServer actually returned for that method and then use that for testing to see if it can be serialized/deserialized properly. I think the same thing was done here in the resource manager. Here is the [thirft service implementation class](https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/resourcemanager/ResourceManagerServer.java) while for the testing class they took a similar approach [here](https://github.com/prestodb/presto/blob/master/presto-main/src/test/java/com/facebook/presto/resourcemanager/TestingResourceManagerClient.java)

The test class in the description (**TestCatalogServerResponse**) will basically test the deserialization phase  which the RemoteMetadataManager covers. If there are any insights into how being able to test directly though I would be glad to hear any opinions

Reply:
Lets discuss this offline, we should be able to test e2e with runner or stubs.

Reply:
Have left a TODO for now in the comments, will be done in a separate PR